build/build_app_engine_package.py

```diff
@@ -95,14 +95,28 @@ def main(src_path, dest_path):
 
   build_version_info_file(os.path.join(dest_path, 'version_info.json'))
 
-if __name__ == '__main__':
+
+def CopyTests(src_path, dest_path):
+  for dirpath, _, files in os.walk(src_path):
+    if dirpath.endswith('app_engine'):
+      tests = [name for name in files if 'test' in name]
+      for test in tests:
+        shutil.copy(os.path.join(dirpath, test), dest_path)
+
+
+def main():
   parser = optparse.OptionParser(USAGE)
+  parser.add_option("-t", "--include-tests", action="store_true",
+                    help='Also copy python tests to the out dir.')
   options, args = parser.parse_args()
   if len(args) != 2:
-    print 'Error: Exactly 2 arguments required.'
-    parser.print_help()
-    sys.exit(1)
+    parser.error('Error: Exactly 2 arguments required.')
 
-  SRC_PATH = args[0]
-  DEST_PATH = args[1]
-  main(SRC_PATH, DEST_PATH)
+  src_path, dest_path = args[0:2]
+  CopyApprtcSource(src_path, dest_path)
+  if options.include_tests:
+    CopyTests(src_path, dest_path)
```

Comment:
So you mean we should remove the tests from the out folder after running them? That sounds pretty ridiculous, and it would be error-prone as well; consider for instance the case when the test crashes half-way through. We would have to be very careful to ensure cleanup in that case.

All the pushing user needs to do is to run grunt build, which will wipe and re-assemble the out folder. That will remove the tests. In fact, we could even make pushing a grunt action so the user don't even have to remember this. Like, grunt push = grunt build + a grunt action which calls dev_appserver.py upload ... We do already have a GAE sdk checked out and Grunt knows where it is, so that should be doable.

Before we do all that, let me challenge your assertion that pushing the tests is super bad. Why? Because it takes a few more kb of uploading? The tests won't get run on the server anyway.